src/db_adapters/blocks.rs

```diff
@@ -53,3 +53,16 @@ pub(crate) async fn latest_block_height(
         .map_err(|err| format!("DB Error: {}", err))?
         .and_then(|(block_height,)| block_height.to_u64()))
 }
+
+/// Gets the latest block before given timestamp (nanos) from database
+pub(crate) async fn closest_block_for(
```

Comment:
While in Rust we can't use named arguments in function call like `closest_block_for(timestamp = 1)` I suggest to add `timestamp` to the function name
```suggestion
pub(crate) async fn closest_block_for_timestamp(
```